lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -133,29 +131,29 @@ def check_assoc(assoc)
           end
         end
 
-        def add_resource_relationships(attrs, serializer, options = {})
-          options[:add_included] = options.fetch(:add_included, true)
-
-          serializer.associations.each do |association|
-            key = association.key
-            serializer = association.serializer
-            opts = association.options
-
-            attrs[:relationships] ||= {}
-
-            if serializer.respond_to?(:each)
-              add_relationships(attrs, key, serializer)
+        def resource_relationship_value(serializer, options = {})
+          if serializer.respond_to?(:each)
+            serializer.map { |s| resource_identifier(s) }
+          else
+            if options[:virtual_value]
```

Comment:
After giving it a thought, why would anyone *ever* use a virtual value for a relationship with json_api? The only value this can take is `nil` or a Resource Identifier/array of Resource Identifiers, which means the virtual attribute has to be either of those, which means the use case is something like:
```ruby
PostSerializer < ActiveModel::Serializer
  attributes :id, :name
  belongs_to :author

  def author
    { type: 'authors', id: '3' }
  end
end
```

which in turn implies that one can't really include the corresponding resource.

Hence, I vote for removing the virtual values for relationships with json_api adapter.